test/pipelinerun_test.go

```diff
@@ -33,12 +34,102 @@ import (
 	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
 )
 
+// dumpAndTearDown will, when a test fails, print out all CRD objects as YAML
+// in the relevant namespace before tearing the namespace down. It will also
+// get the logs of all pods in the namespace. If the tests do not fail then the
+// namespace is simply torn down.
+func dumpAndTearDown(t *testing.T, logger *logging.BaseLogger, cs *clients, namespace string) {
+	defer tearDown(logger, cs.KubeClient, namespace)
+	if !t.Failed() {
+		return
+	}
+
+	var output []byte
+
+	printOrAdd := func(kind, name string, i interface{}) {
+		bs, err := yaml.Marshal(i)
```

Comment:
another option would be using something like `kr.pretty` - not sure that's better tho, output looks great as it is!

Reply:
That is a useful library to know about, but I don't think it would give us anything extra